### ISSUE TITLE
`std.Build.Step.ConfigHeader`: `@cmakedefine X @X@` causes false error when `X` isn't specified

### ISSUE THREAD

**Issue by Lzard** [ziglang/zig#21265]
### Zig Version

0.14.0-dev.1002+71a27ebd8

### Steps to Reproduce and Observed Behavior

In a fresh `zig init` project:
```zig
const std = @import("std");

pub fn build(b: *std.Build) void {
    const config_header = b.addConfigHeader(.{
        .style = .{ .cmake = b.path("src/config_header.h.cmake") },
    }, .{});
    b.getInstallStep().dependOn(&config_header.step);
}
```
With this `src/config_header.h.cmake`:
```c
#cmakedefine X @X@
```
`zig build` causes this error:
```
error: (...)\src\config_header.h.cmake:1: unable to substitute variable: error: MissingValue
```

### Expected Behavior

Lines using the pattern `#cmakedefine X @X@` shouldn't cause an error when `X` isn't specified.

This is caused by `ConfigHeader`'s cmake logic is first scanning for names within `@` `@`. When a name is found without a corresponding value, the error is immediately raised, regardless of whether it is used on a `#cmakedefine` line that depends on its definition.

The following `config_header.h.cmake` builds without issue whether X is specified or not:
```c
#cmakedefine X
```

**Comment by castholm** [comment]
Wouldn't the best and most foolproof behavior be to always require the identifier to be defined, regardless of if takes on the form `#cmakedefine FOO` or `#define FOO @FOO@` or any combination of the two? When I need to configure config headers for my own projects, I usually want to make sure that I have complete coverage of all identifiers present in the input file (using `.FOO_BAR = null` to explicitly undef) and would prefer a build error if I've missed any.

**Comment by Lzard** [comment]
Then the question is whether we want to mirror CMake's behavior, or enforce stricter rules and require values even for `#cmakedefine` lines.

**Comment by robinheydon** [comment]
I would agree with option 1. Although I've just spent a little-bit-of-time to get a few hundred configuration options setup for a project, having to have them all set to `null` places a lot of noise in the way of the signal.

One other little issue (I can raise a separate issue if people think this is a problem) - when you pass in a string as a value in the `values` input, this gets silently changed to an .ident. I'm not sure that this is the correct behaviour, as the generated cmake configuration file surrounds the strings with `"`s.
